src/python/pants/backend/python/goals/coverage_py_integration_test.py

```diff
@@ -315,8 +315,10 @@ def test_coverage_html_xml_json_lcov(batched: bool) -> None:
 def test_default_coverage_issues_12390() -> None:
     # N.B.: This ~replicates the repo used to reproduce this issue at
     # https://github.com/alexey-tereshenkov-oxb/monorepo-coverage-pants.
-    if platform.system() == "Darwin" and platform.processor() == "arm":
```

Comment:
It turns out that `platform.processor()` is an empty string on Linux ARM64, and in general `machine()` is preferred.